ui/components/app/modals/deposit-token-modal/deposit-token-modal.component.js

```diff
@@ -0,0 +1,184 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { EVENT } from '../../../../../shared/constants/metametrics';
+import { NETWORK_TO_NAME_MAP } from '../../../../../shared/constants/network';
+import Button from '../../../ui/button';
+import LogoCoinbasePay from '../../../ui/logo/logo-coinbasepay';
+import LogoTransak from '../../../ui/logo/logo-transak';
+
+export default class DepositEtherModal extends Component {
```

Comment:
We aren't creating new class components, this should be refactored to use functional component pattern, and use hooks instead of a container. The .component and .container suffixes are deprecated, just create a single file that uses hooks for tapping into values in state.

Reply:
hi @brad-decker sorry for the delay on replying, i have been out for the last couple of weeks. idea a couple of weeks ago was to go for the low hanging fruit of adding these without any big changes, so thats why i choose to duplicate the component.

wasn't aware of the intention to deprecate some old code patterns so i used what was there as an example.

ended up working on a single/fits both use cases modal based on the design system components as you suggested. also migrated the ether logic to functional + hooks, since it didnt make sense to keep two separate modals with different code styling for this